utils/misc/misc.go

```diff
@@ -1354,3 +1354,34 @@ func CopyStringMap(originalMap map[string]string) map[string]string {
 	}
 	return newMap
 }
+
+func GetAuthType(config map[string]interface{}) string {
+	var lookupErr error
+	var authValue interface{}
+	if authValue, lookupErr = NestedMapLookup(config, "auth", "type"); lookupErr != nil {
+		return ""
+	}
+	authType, ok := authValue.(string)
+	if !ok {
+		return ""
+	}
+	return authType
+}
+
+type GetAccountIdParams struct {
+	DestConfig map[string]interface{}
+	IdKey      string
+}
+
+func GetAccountId(params GetAccountIdParams) string {
+	idKey := params.IdKey
+	if strings.TrimSpace(idKey) == "" {
```

Comment:
In user delete flow, wouldn't the idKey be `rudderUserDeleteAccountId` to search in DestConfig ?

Reply:
Yes. We are sending the idKey as `rudderDeleteAccountId` **[here](https://github.com/rudderlabs/rudder-server/pull/2632/files#diff-a378f351d7562810ac2b3f21d979516467ca2da122de096005221d3e2218fbf8R246)**. Can you please verify ?

Reply:
Ok, then why to default to `rudderAccountId` ? Will this not effect the eventDelivery account configuration ?

Reply:
When someone doesn't specifically request a specific `idKey` parameter, we assume they're probably asking the account information for event-delivery and provide the information. Is there any problem with this approach ?